### PR TITLE
[Merged by Bors] - fix: disallow untagged or named enum without constant mapping

### PR DESCRIPTION
**Pull request by EstebanBorai**

### CODE REVIEW

File: crates/fluvio-protocol/src/link/error_code.rs [hunk 1]
```diff
@@ -173,10 +173,10 @@ pub enum ErrorCode {
     #[fluvio(tag = 8003)]
     #[error("the derivedstream {0} is invalid")]
     DerivedStreamInvalid(String),
-    #[error("can't do recursive derivedstream yet: {0}->{1}")]
-    DerivedStreamRecursion(String, String),
-    #[error("the derivedstream already exists")]
-    DerivedStreamAlreadyExists,
+    // #[error("can't do recursive derivedstream yet: {0}->{1}")]
```
Comment (galibey) [inline crates/fluvio-protocol/src/link/error_code.rs:176]: Let's remove these commented-out lines.

**Review by galibey — APPROVED** [review]
LGTM.

File: crates/fluvio-protocol/src/link/error_code.rs (hunk 1 above)
Comment (EstebanBorai) [reply crates/fluvio-protocol/src/link/error_code.rs]: Resolved on: 4d4e67f

**Comment by EstebanBorai** [comment]
Clippy issues are being addressed on https://github.com/infinyon/fluvio/pull/3067

**Comment by EstebanBorai** [comment]
@galibey clippy issues where fixed on #3067 and then applied to this PR via rebase

**Review by galibey — APPROVED** [review]

**Review by sehz — APPROVED** [review]
Nice job. Looks good

**Pull request closed without merging**